### PR TITLE
Add simple user-management tool for emailsignin

### LINKED ISSUE #702: Simple Cmdline User-Creation Tool
Hi,

since #272 (User Administration) is a big of a bigger hunk and will need some time, I propose an intermediate step, which will already help users hosting closed instances: A simple commandline user creation/deletion script.

Currently users hosting a closed instance need to manually hash passwords using scrypt on the node commandline, and enter user-data into their respective database. AFAIK no database-commandline or frontend has scrypt-kdf functionality builtin, so this is quite labour-intensive.

A _simple_ Commandline tool that does something like:
```
node user_admin.js
    Usage: node user_admin.js <-a||-d> [-p <password>] <user email>
        -a Add User
        -d Delete User
        -p Use password from commandline (rather than prompting)
```
Would lighten the work of administrators of such a instance tremendously. At the same time, it should be quite simple and quick to implement, so that it can be available quickly.

I could try to understand your database backend and how to `argparse` in node, and implement/PR a functionality like that. Please let me know what you think about the idea.

Best,
-Dario

### PR DESCRIPTION
**Pull request by Nebukadneza**
Fixes #702 

Currently, administrators of closed instances need to manually fiddle in their databases for user-management. This PR adds a small commandline utility that allows to create and
delete users.

I am not very fluid in JavaScript/NodeJS, so I’d be glad for any criticism and advice on my PR. I tried to use async functions, wrapping any functionality i wanted to use synchronously with `await`. This, to me, makes sense in a linearly-progressing commandline utility. I hope that’s good enough style. This also implies that we need node >= 8.x, to have `await` available ….
Also, I was surprised to see that `Instance.destroy()` from `sequelize` does not offer a success return value or similar … so I just to trust it did the right thing.

### CODE REVIEW

**Comment by Nebukadneza** [comment]
Sorry, i can’t seem to find out what Travis’ `spellcheck bin/*` even is, and why it doesn’t like the opening parenthesis of my first require …. Any assistance welcome ^_^.

**Comment by SISheogorath** [comment]
I think there is an exclude parameter we can use ^^ Have to check it myself

**Comment by Nebukadneza** [comment]
Offtopic, but … I’m amazed how I was able to constantly, for 30minutes, misread „shellcheck“ for „spellcheck“, and only notice now. Brain, wow!

**Comment by SISheogorath** [comment]
Actually I was thinking about exactly this solution 👍 

I think there is also an exclude flag but I would also go this way :D

And yes, shellcheck vs. spellcheck happens often to me, too. In another project I had a spellcheck.sh and a shellcheck.sh in the tests directory, debugging was fun sometimes :D

**Comment by SISheogorath** [comment]
Only missing right now (and sadly you have to edit both commits for this) is the signoff. I will test the tool later tonight or tomorrow :)

Thanks a lot for your work!

**Comment by Nebukadneza** [comment]
Sorry, had already forgotten that this is required here. My bad …

File: bin/manage_users [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	await existing_user.destroy();
+	console.log("Deleted user "+argv["del"]+" ...");
+}
+
+// Perform commandline-parsing
+var argv = minimist(process.argv.slice(2));
+
+// Check for add/delete missing
+if (argv["add"] == undefined && argv["del"] == undefined) {
+	console.log("You didn not specify either --add or --del!");
```
Comment (SISheogorath) [inline bin/manage_users]: Some typo: `didn`.

File: bin/manage_users [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	// Find whether we use cmdline or prompt password
+	if(argv["pass"] == undefined) {
+		var pass = readline.question("Password for "+argv["add"]+":", {hideEchoBack: true});
+	} else {
+		console.log("Using password from commandline...");
+		var pass = argv["pass"];
+	}
+
+	// Lets try to create, and check success
+	var ret = await models.User.create({email: argv["add"], password: pass});
```
Comment (SISheogorath) [inline bin/manage_users]: This should be `ref`.

**Comment by Nebukadneza** [comment]
Hi,

rebased onto master again, and fixed your remarks. Thanks for catching, and sorry i haven’t looked well myself …

**Comment by SISheogorath** [comment]
Thanks a lot for your work! It's a great contribution and will help many setups :)

**Comment by bonanza123** [comment]
Thanks so much for this great tool, however, I get the following warning when using it:

> (node:78) DeprecationWarning: Using the automatically created return value from client.query as an event emitter is deprecated and will be removed in pg@7.0. Please see the upgrade guide at https://node-postgres.com/guides/upgrading
> Using password from commandline...

**Comment by ccoenen** [comment]
@bonanza123 please open a new issue for this. Closed / old tickets usually don't get much attention.

Currently that is "only" a warning, so it should work fine. But someone needs to take a look at it, soon, so opening a ticket is a good idea!

**Pull request merged**